PWGGA/GammaConv/AliAODConversionPhoton.cxx

```diff
@@ -265,6 +300,68 @@ void AliAODConversionPhoton::SetCaloPhotonMCFlags(AliMCEvent *mcEvent, Bool_t en
         isShower                      = kTRUE; 
       }
     }
+    // consider the rare case, where the conversion electron stems from photon, which stems from electron, which stems from photon, ...
+    if (isConversion && TMath::Abs(particleGrandMotherPDG) == 11   && particleGrandMotherX2PDG == 22){
+//      printf("gamma -> electron -> gamma -> electron!\n");
+      SetCaloPhotonMCLabel(0,particleGrandMotherLabel);
+      fCaloPhotonMotherMCLabels[0] = particleGrandMotherX3Label;
+      if (TMath::Abs(particleGrandMotherX3PDG) == 11 && particleGrandMotherX4PDG == 22  ){
+//        printf("gamma -> electron -> gamma -> electron -> gamma -> electron!\n");
+        SetCaloPhotonMCLabel(0,particleGrandMotherX3Label);
+        fCaloPhotonMotherMCLabels[0] = particleGrandMotherX5Label;
+        if (TMath::Abs(particleGrandMotherX5PDG) == 11 && particleGrandMotherX6PDG == 22  ){
+//          printf("gamma -> electron -> gamma -> electron -> gamma -> electron -> gamma -> electron!\n");
+          SetCaloPhotonMCLabel(0,particleGrandMotherX5Label);
+          fCaloPhotonMotherMCLabels[0] = particleGrandMotherX7Label;
+        }
+      }
+    }
+
+    // consider the case, where a photon stems from a photon which stems from a photon etc...which is common for frag. photons
+    TParticle *dummyMother = mcEvent->Particle(particleMotherLabel);
```

Comment:
The following line causes a segfault if particleMotherLabel was "-1", please always tet for the pointer to actually exist.